histomicsui/web_client/panels/MetadataWidget.js

```diff
@@ -0,0 +1,529 @@
+        }
+    },
+
+    setItem: function (item) {
+        this.item = item;
+        this.item.on('g:changed', function () {
+            this.render(0);
+        }, this);
+        this.render(0);
+        return this;
+    },
+
+    // Does not support modal editing
+    getModeFromValue: function (value) {
+        return _.isString(value) ? 'simple' : 'json';
+    },
+
+    addMetadata: function (event, mode) {
+        var EditWidget = this.modes[mode].editor;
+        var value = (mode === 'json') ? '{}' : '';
+        // expand the widget when adding new metadata
+        this.renderMetadataWidgetHeader();
+        var widget = new MetadatumWidget({
+            className: 'g-widget-metadata-row editing',
+            mode: mode,
+            key: '',
+            value: value,
+            item: this.item,
+            fieldName: this.fieldName,
+            apiPath: this.apiPath,
+            accessLevel: this.accessLevel,
+            parentView: this,
+            onMetadataEdited: this.onMetadataEdited,
+            onMetadataAdded: this.onMetadataAdded
+        });
+        widget.$el.appendTo(this.$('.g-widget-metadata-container'));
+
+        new EditWidget({
+            item: this.item,
+            key: '',
+            value: value,
+            fieldName: this.fieldName,
+            apiPath: this.apiPath,
+            accessLevel: this.accessLevel,
+            newDatum: true,
+            parentView: widget,
+            onMetadataEdited: this.onMetadataEdited,
+            onMetadataAdded: this.onMetadataAdded
+        })
+            .render() //
+            .$el.appendTo(widget.$el);
+    },
+
+    renderMetadataWidgetHeader: function () {
+        // pervent automatically collapse the widget after rendering again
+        this.render(1);
+    },
+
+    render: function (e) {
+        // e is a variable that determines how to render the widget
+        if (this.item && this.item.id) {
+            const imageId = this.item.id;
+            const apiPath = `/item/${imageId}/metadata`;
+            this.item.getAccessLevel((accessLevel) => {
+                var metaDict = this.item.get(this.fieldName) || {};
+                var metaKeys = Object.keys(metaDict);
+                metaKeys.sort(localeSort);
+                // Metadata header
+                var firstKey = (metaKeys)[0];
+                var firstValue = metaDict[firstKey];
+                if (_.isObject(firstValue)) {
+                    // if the value is a json object, JSON.stringify to make it more readable
+                    firstValue = JSON.stringify(firstValue);
+                }
+                this.$el.html(metadataWidgetTemplate({
+                    item: this.item,
+                    title: this.title,
+                    firstKey: firstKey,
+                    firstValue: firstValue,
+                    accessLevel: this.item.attributes._accessLevel,
+                    AccessType: AccessType
+                }));
```

Comment:
I notice when we first load a page with metadata, that the metadata panel is open and quickly closes.  It would be better to start it in the state we want.  Rather than tracking how many times a render has occurred, we can dispense with that variable and just pass an appropriate value to the settings of the MetadaumWidget:
```
collapsed: !this.$('.s-panel-content').is(':visible')
```
The logic here is that if we've never rendered, the jquery selector will be empty and hence "not visible"; otherwise, we are maintaining the current state.

Reply:
OK, thank you David.